### ISSUE TITLE
emby/jellyfin support

### ISSUE THREAD

**Issue by binarygeek119** [petersem/posterr#49]
emby/jellyfin are built for the same source when emby went closed source  jellyfin was born.
if you add support for one the other be easy to adept to.
jellyfin is free 100%
emby is like plex
right now i run both emby and plex for other software but once they get emby/jellyfin support i will drop plex.

emby api
https://github.com/MediaBrowser/Emby/wiki

jellyfin api
https://jellyfin.org/docs/plugin-api/index.html

**Comment by petersem** [comment]
The plan, since day one, has been to add this. Classes are already in the code as placeholders. I run Jellyfin also.

**Comment by Darkwolf1515** [comment]
That's fantastic to hear! There was a similar project (that sadly wasn't open source) for displaying plex images from a pi, while I asked for jellyfin and he was interested, nothing came from it, I cannot wait to see it here!

**Comment by petersem** [comment]
@Darkwolf1515 It will all depend on what the data and API is like. Unfortunately, the Jellyfin API seems to have no formally published doco. I'll look at this post v1, and where other enhancements with higher priority have been implemented. Feel free to send me any JF API info that you can find.

**Comment by Darkwolf1515** [comment]
Take a look at this https://github.com/jasongdove/ErsatzTV/releases/tag/v0.0.41-prealpha he got Jellyfins API to pull posters and much much more.

**Comment by petersem** [comment]
@Darkwolf1515 interesting. effectively i'd have to write the whole API again in a node package to be compatible. But it does give me some ideas. There s possibly an already exisiting jellyfin node package out there. I havent even done a preliminary look as yet. I know officially, theres no doco

**Comment by Darkwolf1515** [comment]
I asked him and he said he also got his docs from `http://your-jellyfin-server:8096/api-docs/swagger/index.html`

**Comment by Darkwolf1515** [comment]
https://api.jellyfin.org/ and here too!

**Comment by acvigue** [comment]
> @Darkwolf1515 interesting. effectively i'd have to write the whole API again in a node package to be compatible. But it does give me some ideas. There s possibly an already exisiting jellyfin node package out there. I havent even done a preliminary look as yet. I know officially, theres no doco

If you're using Axios, there's a generated client in the Jellyfin org. https://github.com/jellyfin/jellyfin-client-axios 📦

**Comment by petersem** [comment]
Thanks, I'll have a look

**Comment by petersem** [comment]
I'm planning to stark working on jellyfin integration in the next day or so. Will do a bit of proof of concept tasks to ensure I can get the data, then see how well it matches up with my base classes. If that works, I'm hoping it won't take too long to do

**Comment by petersem** [comment]
OK, so this should be easy, but it's turning out to be tricker than expected. Whilst there is a simple API call to show me what is playing, it doesnt show every title playing. Ive been picking apart the jellyfin-web code, trying to understand what it is doing. The API is quite a pain. :( Any suggestions on a single api call that will return all playing media items?

**Comment by Bitwolfies** [comment]
Id try asking in the jellyfin matrix/discord/irc (its all the same chat)

**Comment by petersem** [comment]
> Id try asking in the jellyfin matrix/discord/irc (its all the same chat)

Yeah, tried discord and Reddit, however no tangible assistance. ☹

**Comment by Bitwolfies** [comment]
Maybe it’s for the best, iirc jf will soon undergo a massive API rewrite.

**Comment by petersem** [comment]
ill get there. is just frustrating that it isnt well documented or anywhere as simplae as the plex api is

**Comment by Elfman72** [comment]
Looks like we are going over a year now but just found this project.  Would also love an Emby/Jellyfin integration for this if at all possible.

Thanks!

**Comment by tcj-one** [comment]
Bummer this was difficult to implement, I'm redoing my whole home server stack and will be using Jellyfin instead of Plex this time, and I'd love to maintain my Posterrs.

**Comment by Blu3wolf** [comment]
> emby api https://github.com/MediaBrowser/Emby/wiki
> 
> jellyfin api https://jellyfin.org/docs/plugin-api/index.html

The new link is https://api.jellyfin.org/

**Comment by joobert** [comment]
Bump. This is needed!

**Comment by Blu3wolf** [comment]
https://jmshrv.com/posts/jellyfin-api/ - looks like a handy intro. 
https://api.jellyfin.org/#tag/Session/operation/GetSessions - this looks like the API call you'd want?

**Comment by or-agami** [comment]
@petersem Any plans to implement?

**Comment by gurkburk76** [comment]
friendly bump, saying to to plex and yes  to jellyfin in my mancave :)

**Comment by 12nick12** [comment]
Any update? I wish I was smart enough to contribute.

**Comment by binarygeek119** [comment]
hey @petersem i was able to get chatgpt to add jellyfin support for posterbox.
you may be able to use that info for your app.
i couldnt get chatgpt to add jellyfin support for posterr. so i looked to work with another project to see if it could be done.
https://github.com/binarygeek119/PosterBox-Jellyfin

**Comment by binarygeek119** [comment]
i got a ai edited version of poster working with jellyfin now
https://github.com/binarygeek119/posterr
i also add a few new thing to it aswell
emby 
kodi
lidarr
music 
artist actor director images
book support - jellyfin readarr/chaptarr
chaptarr support or readarr
ect.
it is all ai edited thought so use at your own risk. but jellyfin is working
emby and kodi is untested
 
real dev work is welcome to take over this and overhaul it with human written code.
im not a dev.
